internal/karma/karma.conf.js

```diff
@@ -3,7 +3,57 @@
 try
 {
   const fs = require('fs');
+  const path = require('path');
   const tmp = require('tmp');
+  const child_process = require('child_process');
+  const isWindows = /^win/i.test(process.platform);
+
+  // Helper function to find a particular namedFile
+  // within the webTestMetadata webTestFiles
+  function findNamedFile(webTestMetadata, key) {
```

Comment:
seems like we could remove all the helper code for the extract_exe if we always unzip ourselves?

Reply:
same as above: that was my original plan. however on Linux the archive is a tar.xz which adm-zip doesn't handle

Reply:
what about https://www.npmjs.com/package/xz + https://www.npmjs.com/package/tar ?

I don't love adding external dependencies but also this could be a lot simpler, WDYT?

Reply:
Yea. Using the provided EXTRACT_EXE is not idea. I'll try those packages out.

Reply:
I thought about this some more and looked at the npm packages. I think we'll run into trouble supporting the various compression formats in node. On Windows we can assume it will be a `.zip` file. On linux you may get `.tar.bz2` (which is what it is currently actually and not tar.xz) or `tar.gz` or `tar.xz` in the future. If we stick to using EXTRACT_EXE then it ensures we can always extract on Linux and OSX:
```
if [[ "${BASENAME}" == *.deb ]]; then
  wrap dpkg -x "${ARCHIVE}" "${OUT_DIR}"
elif [[ "${BASENAME}" == *.tar ]]; then
  wrap tar xf "${ARCHIVE}" -C "${OUT_DIR}"
elif [[ "${BASENAME}" == *.tar.bz2 || "${BASENAME}" == *.tbz2 ]]; then
  wrap tar xjf "${ARCHIVE}" -C "${OUT_DIR}"
elif [[ "${BASENAME}" == *.tar.gz || "${BASENAME}" == *.tgz ]]; then
  wrap tar xzf "${ARCHIVE}" -C "${OUT_DIR}"
elif [[ "${BASENAME}" == *.tar.Z ]]; then
  wrap tar xZf "${ARCHIVE}" -C "${OUT_DIR}"
elif [[ "${BASENAME}" == *.zip ]]; then
  wrap unzip "${ARCHIVE}" -d "${OUT_DIR}"
else
  >&2 printf "${ARCHIVE} is not a supported file type."
  exit -1
fi
```

Reply:
Okay I think @DrMarcII might be interested, seems like EXTRACT EXE should be fixed in windows

Reply:
@DrMarcII Ahhh. That's good to know. I didn't notice that. Can use EXTRACT_EXE on all platforms then 👍